eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/RawExceptionThrowCheck.java

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FullIdent;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import static com.puppycrawl.tools.checkstyle.utils.TokenUtil.findFirstTokenByPredicate;
+
+/**
+ * Check that verifies that exceptions are logged when they are created.
+ * The check is skipped if:
+ * - the exception is created in a generated method or service interface
+ * - the exception is a NullPointerException, IllegalArgumentException or UnsupportedOperationException that are used for immediate input validation
+ */
+public class RawExceptionThrowCheck extends AbstractCheck {
+    static final String ERROR_MESSAGE = "Directly throwing a new exception is disallowed. Use the \"io.clientcore.core.instrumentation.logging.ClientLogger\" API instead, "
+        + "such as \"logger.throwableAtError()\" or \"logger.throwableAtWarning()\"."
+        + "See https://github.com/Azure/azure-sdk-for-java/wiki/Client-core:-logging-exceptions-best-practices for more details.";
+
+    private static final Set<String> IGNORED_EXCEPTIONS = new HashSet<>(Arrays.asList("NullPointerException",
```

Comment:
Same case as before, since this is non-modifiable and the only usage is to be iterated over this should be a list or an array.